lib/internal/url.js

```diff
@@ -239,8 +239,10 @@ class URL {
   constructor(input, base) {
     // toUSVString is not needed.
     input = `${input}`;
-    if (base !== undefined && !(base instanceof URL))
+    if (base !== undefined &&
+        (!base[searchParams] || !base[searchParams][searchParams])) {
```

Comment:
@mscdex ... is there benefit to moving this into a separate `isURL(u)` function that can be exported by `internal/url`?